### ISSUE TITLE
bug: uneven number of samples in wave_python

### ISSUE THREAD

**Issue by mvonpapen** [chris-torrence/wavelets#1]
Dear Chris,

I think I may have stumbled over a minor bug in the python
function **waveletFunctions.py**: If you run the code on a time series of uneven
length (e.g. `len(x)=101`), it returns "*ValueError: operands could not be
broadcast together with shapes (101,) (100,)*" as the Fourier transform f
and the daughter have different sizes. You may check this via

```
import numpy as np
from waveletFunctions import wavelet

x = np.random.randn(101)
dt = 1.0
w = wavelet ( x, dt )
```

The problem is in the construction of the wave vector k. Here, the last
entry of the `kplus` vector is always omitted (`kplus[0:-1]`). However, in
case of uneven length of the time series, the Fourier transform returns
no value for the Nyquist frequency and therefore the last entry of wave
vector kplus should be used as well. I found that the following lines
replacing **lines 115-119** solve this problem:

```
#....construct wavenumber array used in transform [Eqn(5)]
kplus = np.arange(1, n/2+1)
kplus = (kplus * 2 * np.pi / (n * dt))
kminus = (-(kplus[0:(n-1)/2])[::-1])
k = np.concatenate(([0.], kplus, kminus))
```

Here, I simply replaced `kplus[0:-1]` by `kplus[0:(n-1)/2]`. The np.fix
function is actually not needed as dividing by integer automatically
takes care of rounding down.

Best regards,
Mitch

**Comment by ct6502** [comment]
Merged your code - thanks so much!

**Comment by khider** [comment]
Encountered the same problem as above with the proposed changes. When n=2001 and dt=1, k is 2002.

I modified as follows:

```
kplus = np.arange(1, int(n / 2) + 1)
kplus = (kplus * 2 * np.pi / (n * dt))
kminus = np.sort(-kplus) 
k = np.concatenate(([0.], kplus, kminus))
```

This gives me a similar k as the Matlab code (which I used as a ground truth basis).

**Comment by ct6502** [comment]
> Encountered the same problem as above with the proposed changes. When n=2001 and dt=1, k is 2002.
> 
> I modified as follows:
> 
> ```
> kplus = np.arange(1, int(n / 2) + 1)
> kplus = (kplus * 2 * np.pi / (n * dt))
> kminus = np.sort(-kplus) 
> k = np.concatenate(([0.], kplus, kminus))
> ```
> 
> This gives me a similar k as the Matlab code (which I used as a ground truth basis).

Thanks @khider - does your new code also work for even "n" as well? If so, would it be possible for you to do a pull request and I can merge it in?

**Comment by khider** [comment]
Just updated to work for even and uneven as well as adding a couple of lines in case the frequency vector is set to None. I did a pull request.

**Comment by ct6502** [comment]
Great, thanks! Just merged the PR #5 . Closing again.
